src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -4635,6 +4836,10 @@ class StubGenerator: public StubCodeGenerator {
       StubRoutines::_montgomerySquare = g.generate_square();
     }
 
+    if (UsePoly1305Intrinsics) {
```

Comment:
Why is it in the `#ifdef COMPILER2` block? What's the dependency on C2? If it's not necessary, you can move it to https://github.com/openjdk/jdk/pull/16417/files?diff=unified&w=0#diff-97f199af6d1c8c17b2fa4f50eb1bbc0081858cc59a899f32792a2d31f933ccc4R4861.

Reply:
So far this intrinsic can be used by C2 only, so it's clever to not compile this code if we build jdk without c2.